lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java

```diff
@@ -26,44 +26,75 @@
 public class XmlReader {
     private static final Logger LOGGER = Logger.getLogger(XmlReader.class.getName());
 
-    public static boolean hasServerRoot(File file) throws XMLStreamException, IOException {
+    public static boolean hasServerRoot(File file) {
         if (!file.exists() || file.length() == 0) {
             return false;
         }
+
         XMLInputFactory factory = XMLInputFactory.newInstance();
-        XMLEventReader reader = factory.createXMLEventReader(new FileInputStream(file));
-        if (reader.hasNext()) {
-            XMLEvent firstTag = reader.nextTag(); // first start/end element
-            reader.close();
-            return isServerElement(firstTag);
+        XMLEventReader reader = null;
+        try {
```

Comment:
You slacked me and I responded, but guess you did not see. You can do a try/finally without a catch, so that you can keep the model of throwing the exceptions on the method signature.